src/modules/synchronizer/GitHubSynchronizer.js

```diff
@@ -0,0 +1,170 @@
+        repositoryId: repository.id,
+      })
+
+      if (!userRepositoryRight) {
+        await UserRepositoryRight.query().insert({
+          userId: this.synchronization.user.id,
+          repositoryId: repository.id,
+        })
+      }
+    }))
+
+    if (this.github.hasNextPage(githubRepositories)) {
+      await this.synchronizeRepositories({ page: page + 1 })
+    }
+  }
+
+  async synchronizeOwners(githubRepositories, type) {
+    const githubOwners = githubRepositories
+      .reduce((githubOwners, githubRepository) => {
+        if (githubRepository.owner.type !== type) {
+          return githubOwners
+        }
+
+        let githubOwner = githubOwners.find(({ id }) => id === githubRepository.owner.id)
+
+        if (!githubOwner) {
+          githubOwner = githubRepository.owner
+          githubOwners.push(githubRepository.owner)
+        }
+
+        return githubOwners
+      }, [])
+
+    let owners
+
+    switch (type) { // eslint-disable-line default-case
+      case OWNER_ORGANIZATION:
+        owners = await Promise.all(
+          githubOwners.map(githubOwner => this.synchronizeOrganization(githubOwner)),
+        )
+        break
+      case OWNER_USER:
+        owners = await Promise.all(
+          githubOwners.map(githubOwner => this.synchronizeUser(githubOwner)),
+        )
+        break
+    }
+
+    return githubRepositories
+      .reduce((ownerIdByRepositoryId, githubRepository) => {
+        if (githubRepository.owner.type === type) {
+          ownerIdByRepositoryId[githubRepository.id] = owners
+            .find(owner => owner.githubId === githubRepository.owner.id).id
+        }
+
+        return ownerIdByRepositoryId
+      }, {})
+  }
+
+  async synchronizeOrganization(githubOrganization) {
+    let [organization] = await Organization.query().where({ githubId: githubOrganization.id })
+    const data = {
+      githubId: githubOrganization.id,
+      name: githubOrganization.login,
+    }
+
+    if (organization) {
+      await organization.$query().patch(data)
+    } else {
+      organization = await Organization.query().insert(data)
+    }
+
+    const [userOrganization] = await UserOrganizationRight.query().where({
+      userId: this.synchronization.user.id,
+      organizationId: organization.id,
+    })
+
+    if (!userOrganization) {
+      await UserOrganizationRight.query().insert({
+        userId: this.synchronization.user.id,
+        organizationId: organization.id,
+      })
+    }
+
+    return organization
+  }
+
+  async synchronizeUser(githubUser) { // eslint-disable-line class-methods-use-this
+    let [user] = await User.query().where({ githubId: githubUser.id })
+    const data = { githubId: githubUser.id, name: githubUser.login }
+
+    if (user) {
+      await user.$query().patch(data)
+    } else {
+      user = await User.query().insert(data)
+    }
+
+    return user
+  }
+
+  async synchronize() {
+    await this.synchronization.$relatedQuery('user')
```

Comment:
Be sure that we load the user before starting sync, it is used in the synchronization process.